### PR TITLE
Improve performance of Trame camera sync

### PR DESCRIPTION
**Pull request by banesullivan**
Some performance improvements to the Trame viewer.

This solves https://github.com/pyvista/pyvista/discussions/4055 to improve the camera controls when there are many actors in the scene

I need to do a bit more validation testing on these changes to make the the `update()` -> `update_camera()` change on `render()` calls doesn't break any behavior

### CODE REVIEW

**Comment by banesullivan** [comment]
> I need to do a bit more validation testing on these changes to make the the update() -> update_camera() change on render() calls doesn't break any behavior

This breaks some animation workflows. I will create a validation example and fix

File: examples_trame/validation/ManyActors.py
Comment (akaszynski) [inline examples_trame/validation/ManyActors.py:1]: Should we make these file names snakecase?

Comment (banesullivan) [reply examples_trame/validation/ManyActors.py:1]: Yeah.... I should and will.

Comment (banesullivan) [reply examples_trame/validation/ManyActors.py:1]: Okay done. Except I left three files CamelCase as they are examples demonstrating specific classes

**Review by akaszynski — APPROVED** [review]
LGTM

**Pull request merged**